src/template.rs

```diff
@@ -2,29 +2,86 @@ use home::home_dir;
 
 use std::env;
 use std::fs;
-use std::io;
 use std::path::PathBuf;
+use std::process::Command as OsCommand;
 
-#[derive(Debug)]
-pub enum Error {
-    CouldNotGetHomeDirectory,
-    Other(io::Error),
-}
-
-pub fn get_template_path() -> Result<PathBuf, Error> {
+pub fn get_template_path() -> Result<PathBuf, String> {
     let default_template_path = home_dir()
-        .map(|x| x.join(".bonnie").join("template.toml"))
-        .ok_or(Error::CouldNotGetHomeDirectory)?;
+        .map(|path| path.join(".bonnie").join("template.toml"))
+        .ok_or(format!(
+            "I could not find your home directory. {}",
+            if cfg!(target_os = "windows") {
+                "That is most odd."
+            } else {
+                "Is the `HOME` environment variable set?"
+            }
+        ))?;
```

Comment:
```suggestion
        .ok_or(
            String::from("Your home directory couldn't be found. Please check your system configuration.")
        )?;
```

That shouldn't happen on any system, I think a generic error is fine.

Reply:
Specificity is always helpful, and `HOME` *could* not be set.

Reply:
If `HOME` isn't set, the user is almost certainly running some cursed system! Also, we don't *know* that that's the problem. I think in a system error like this we should leave it open.